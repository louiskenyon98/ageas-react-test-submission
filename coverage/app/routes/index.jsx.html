<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/routes/index.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">app/routes</a> index.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/1</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import {Router, Route, Switch} from 'react-router-dom';
import Navigation from '../components/common/Navigation/Navigation';
import history from '../history';
&nbsp;
import AllNotesContainer from '../containers/notes/AllNotes'
import NoteCreateContainer from '../containers/notes/NoteCreate';
import NoteEditContainer from '../containers/notes/NoteEdit';
&nbsp;
export class Routes extends React.Component {
    render() {
        return (
            &lt;Router history={history}&gt;
                &lt;React.Fragment&gt;
                    &lt;Navigation/&gt;
                    &lt;Switch&gt;
                        &lt;Route path="/edit/:id" component={NoteEditContainer}/&gt;
                        &lt;Route path="/new" component={NoteCreateContainer}/&gt;
                        &lt;Route path="/" component={AllNotesContainer}/&gt;
                    &lt;/Switch&gt;
                &lt;/React.Fragment&gt;
            &lt;/Router&gt;
        )
    }
}
&nbsp;
export default Routes;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Apr 15 2019 14:58:54 GMT+0100 (GMT Daylight Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
