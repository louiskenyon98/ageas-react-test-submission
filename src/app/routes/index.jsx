import React from 'react';
import {Router, Route, Switch} from 'react-router-dom';
import Navigation from '../components/common/Navigation/Navigation';
import history from '../history';

import AllNotesContainer from '../containers/notes/AllNotes'
import NoteCreateContainer from '../containers/notes/NoteCreate';
import NoteEditContainer from '../containers/notes/NoteEdit';

class Routes extends React.Component {
    render() {
        return (
            <Router history={history}>
                <React.Fragment>
                    <Navigation/>
                    <Switch>
                        <Route path="/edit/:id" component={NoteEditContainer}/>
                        <Route path="/new" component={NoteCreateContainer}/>
                        <Route path="/" component={AllNotesContainer}/>
                    </Switch>
                </React.Fragment>
            </Router>
        )
    }
}

export default Routes;
