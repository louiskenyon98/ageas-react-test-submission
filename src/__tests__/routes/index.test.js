import React from 'react';
import {shallow} from 'enzyme'
import {Routes} from '../../app/routes/index';

describe('Routes', () => {
    let props = {};
    let wrapper = () => shallow(<Routes {...props}/>);

    describe('render', () => {
        it('should render Routes component', () => {
            expect(wrapper()).toMatchSnapshot();
        })
    })
});
