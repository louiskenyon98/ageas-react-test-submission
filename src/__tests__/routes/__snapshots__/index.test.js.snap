// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes render should render Routes component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Navigation />
  <Switch>
    <Route
      component={[Function]}
      path="/edit/:id"
    />
    <Route
      component={[Function]}
      path="/new"
    />
    <Route
      component={[Function]}
      path="/"
    />
  </Switch>
</Router>
`;
